main.go

```diff
@@ -16,11 +16,117 @@
 package main
 
 import (
+	"bytes"
+	"context"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/hex"
+	"errors"
 	"fmt"
+	"io/ioutil"
 	"log"
 	"net/http"
+	"os"
+	"strings"
+
+	"github.com/go-openapi/runtime"
+	"github.com/google/go-github/v42/github"
+	"github.com/google/trillian/merkle/logverifier"
+	"github.com/google/trillian/merkle/rfc6962"
+	"github.com/rhysd/actionlint"
+	"github.com/sigstore/cosign/cmd/cosign/cli/options"
+	"github.com/sigstore/cosign/cmd/cosign/cli/rekor"
+	"github.com/sigstore/cosign/pkg/cosign"
+	"github.com/sigstore/cosign/pkg/cosign/bundle"
+	"github.com/sigstore/cosign/pkg/cosign/tuf"
+	"github.com/sigstore/rekor/pkg/generated/client"
+	"github.com/sigstore/rekor/pkg/generated/client/entries"
+	"github.com/sigstore/rekor/pkg/generated/models"
+	"github.com/sigstore/rekor/pkg/types"
+	hashedrekord "github.com/sigstore/rekor/pkg/types/hashedrekord/v0.0.1"
+	rekord "github.com/sigstore/rekor/pkg/types/rekord/v0.0.1"
+	"github.com/sigstore/sigstore/pkg/cryptoutils"
 )
 
+// verifySignature receives the scorecard analysis payload, looks up its associated tlog entry and
+// certificate, and extracts the repository's workflow file to ensure its legitimacy.
+func verifySignature(w http.ResponseWriter, r *http.Request) {
+	ctx := context.Background()
+	reqBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, "error reading http request body", http.StatusInternalServerError)
+		return
+	}
+
+	// Get most recent Rekor entry uuid.
+	rekorClient, err := rekor.NewClient(options.DefaultRekorURL)
+	if err != nil {
+		http.Error(w, "error initializing Rekor client", http.StatusInternalServerError)
+		return
+	}
+	uuids, err := cosign.FindTLogEntriesByPayload(ctx, rekorClient, reqBody)
+	if err != nil || len(uuids) == 0 {
+		http.Error(w, "error fetching tlog entries", http.StatusInternalServerError)
+		return
+	}
+	uuid := uuids[len(uuids)-1] // ignore past entries.
+
+	// Verify tlog entry to make sure it is actually in the log.
+	entry, err := verifyTLogEntry(ctx, rekorClient, uuid)
+	if err != nil {
+		http.Error(w, "error verifying tlog entry", http.StatusInternalServerError)
+		return
+	}
+
+	// Extract certificate and get repo reference & path.
+	certs, err := extractCerts(entry)
+	if err != nil || len(certs) == 0 {
+		http.Error(w, "error extracting certificate from entry", http.StatusInternalServerError)
+		return
+	}
+	cert := certs[0]
+	var repoRef string
+	var repoPath string
+	for _, ext := range cert.Extensions {
+		if ext.Id.String() == "1.3.6.1.4.1.57264.1.6" {
+			repoRef = string(ext.Value)
+		}
+		if ext.Id.String() == "1.3.6.1.4.1.57264.1.5" {
+			repoPath = string(ext.Value)
+		}
+	}
+
+	if len(repoRef) == 0 || len(repoPath) == 0 {
+		http.Error(w, "error extracting repo ref or path from certificate", http.StatusInternalServerError)
+		return
+	}
+
+	// Split repo path into owner and repo name.
+	ownerName := repoPath[:strings.Index(repoPath, "/")]
```

Comment:
you can use `strings.SplitN` or `strings.Split` as well, which may be clearer for reader